### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by helrond**
- Add deploy, test and branch enforcement workflows
- Update dependency workflow
- Remove Travis CI configs

**NOTES**
- Need to install dependencies in this package, so I added a step for that in the deploy workflow.
- The above required the addition of a step to set up a python version

### CODE REVIEW

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -14,9 +14,17 @@ jobs:
       contents: write
       pull-requests: write
     steps:
+      - uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.APP_ID }}
+          private-key: ${{ secrets.PRIVATE_KEY }}
+          
       - uses: actions/checkout@v3
```
Comment (HaSistrunk) [inline .github/workflows/dependencies.yml]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/dependencies.yml [hunk 2]
```diff
@@ -14,9 +14,17 @@ jobs:
...
         with:
+          token: ${{ steps.app-token.outputs.token }}
           ref: development
+          persist-credentials: false
 
       - name: Set up Python
         uses: actions/setup-python@v4
```
Comment (HaSistrunk) [inline .github/workflows/dependencies.yml]:
```suggestion
        uses: actions/setup-python@v5
```

**Review by HaSistrunk — CHANGES_REQUESTED** [review]

**Review by HaSistrunk — APPROVED** [review]

**Review by ioduok — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - development
```
Comment (p-galligan) [inline .github/workflows/deploy.yml:6]: On review, we don't have to deploy on push to dev. Nothing is different between dev and base, and it's only getting deployed once.

I'd remove this line.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          role-to-assume: ${{ secrets.AWS_DEPLOY_ROLE }}
+          role-skip-session-tagging: true
+          role-duration-seconds: 900
+
+      - name: Upload file to S3
+        run: aws s3 cp $LAMBDA_NAME.zip s3://$LAMBDA_BUCKET_NAME/$LAMBDA_NAME.zip
+
+      - name: Deploy lambda package
+        run: aws lambda update-function-code
+          --function-name arn:aws:lambda:$AWS_REGION:$AWS_LAMBDA_ACCOUNT_ID:function:$LAMBDA_NAME
```
Comment (p-galligan) [inline .github/workflows/deploy.yml:63]: Change `$AWS_LAMBDA_ACCOUNT_ID` to `${{ secrets.AWS_LAMBDA_ACCOUNT_ID }}

**Review by p-galligan — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_DEPLOY_ROLE }}
+          role-skip-session-tagging: true
+          role-duration-seconds: 900
+
+      - name: Upload file to S3
+        run: aws s3 cp $LAMBDA_NAME.zip s3://$LAMBDA_BUCKET_NAME/$LAMBDA_NAME.zip
```
Comment (p-galligan) [inline .github/workflows/deploy.yml:59]: `$LAMBDA_BUCKET_NAME` should be `${{ secrets.LAMBDA_BUCKET_NAME }}`

**Review by p-galligan — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (p-galligan) [inline .github/workflows/deploy.yml:6]: We don't have a dev lambda bucket, so I'd say we should just push once, to prod.

**Review by p-galligan — APPROVED** [review]

**Pull request merged**